lib/oc-tests/new_locking/test_uploadFilePlusAction.py

```diff
@@ -0,0 +1,276 @@
+__doc__ = """
+Test locking feature. The test will upload a file big enough and perform an operation
+while the upload is running. Each test set will run a different operation.
+Chunked uploads are currently outside of the tests
+
+Test sets:
+* upload + upload (same file)
+* upload + delete folder
+* upload + rename folder
+* upload + move folder
+
++-------------+----------------------------+--------------------+
+| step number | uploader                   | doer               |
++-------------+----------------------------+--------------------+
+| 2           | upload big file            | create working dir |
++-------------+----------------------------+--------------------+
+| 3           | overwrite big file (async) |                    |
++-------------+----------------------------+--------------------+
+| 4           |                            | perform action     |
++-------------+----------------------------+--------------------+
+| 5           | check result and cleanup   |                    |
++-------------+----------------------------+--------------------+
+| 6           |                            | check result       |
++-------------+----------------------------+--------------------+
+"""
+
+import os
+import re
+import tempfile
+import uuid
+import owncloud
+import zipfile
+import smashbox.utilities.pyocclient_wrapper
+from smashbox.utilities import *
+from smashbox.script import config as sconf
+
+def check_local_filesize(localpath, size):
+    '''username and password remains to keep the expected signature'''
+    return os.path.getsize(localpath) == size
+
+def check_zip_contents(localpath, content_list):
+    checked_result = False
+    with zipfile.ZipFile(localpath, 'r') as myzip:
+        checked_result = myzip.namelist() == content_list
+    return checked_result
+
+def parse_worker_number(worker_name):
+    match = re.search(r'(\d+)$', worker_name)
```

Comment:
I'll make the change after merging to the branch. Download and overwrite tests will need the same change, so I'll do it at the same time